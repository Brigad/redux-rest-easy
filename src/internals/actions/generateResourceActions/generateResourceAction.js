const generateResourceAction = (actionType, resourceName) => resourceId => ({
  type: actionType,
  resourceName,
  resourceId,
});

export default generateResourceAction;
