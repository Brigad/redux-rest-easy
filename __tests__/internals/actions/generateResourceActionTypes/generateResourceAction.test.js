import generateResourceAction from '../../../../src/internals/actions/generateResourceActions/generateResourceAction';

const ACTION_TYPE = '@@rest-easy/fruits/eat/REQUEST';
const RESOURCE_NAME = 'fruits';
const RESOURCE_ID = 2;

describe('generateResourceAction', () => {
  test('no args', () => {
    expect(generateResourceAction()()).toMatchSnapshot();
  });

  test('no resourceName, resourceId', () => {
    expect(generateResourceAction(ACTION_TYPE)()).toMatchSnapshot();
  });

  test('no resourceId', () => {
    expect(
      generateResourceAction(ACTION_TYPE, RESOURCE_NAME)(),
    ).toMatchSnapshot();
  });

  test('happy path', () => {
    expect(
      generateResourceAction(ACTION_TYPE, RESOURCE_NAME)(RESOURCE_ID),
    ).toMatchSnapshot();
  });
});
