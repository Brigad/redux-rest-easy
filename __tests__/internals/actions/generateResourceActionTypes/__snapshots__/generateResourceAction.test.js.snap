// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateResourceAction happy path 1`] = `
Object {
  "resourceId": 2,
  "resourceName": "fruits",
  "type": "@@rest-easy/fruits/eat/REQUEST",
}
`;

exports[`generateResourceAction no args 1`] = `
Object {
  "resourceId": undefined,
  "resourceName": undefined,
  "type": undefined,
}
`;

exports[`generateResourceAction no resourceId 1`] = `
Object {
  "resourceId": undefined,
  "resourceName": "fruits",
  "type": "@@rest-easy/fruits/eat/REQUEST",
}
`;

exports[`generateResourceAction no resourceName, resourceId 1`] = `
Object {
  "resourceId": undefined,
  "resourceName": undefined,
  "type": "@@rest-easy/fruits/eat/REQUEST",
}
`;
